### DISCUSSION TITLE
Integer division not working as expected

### DISCUSSION THREAD

**Discussion by sami043** [duckdb/duckdb#8492] [Q&A]
When running this query, I would expect an answer of 0.5 but I get 0.0:
`duckdb.sql('SELECT CAST(1.0 AS HUGEINT) / CAST(2.0 AS HUGEINT)').df()`

In order to get 0.5 I have to change either the numerator or denominator to a float. What am I missing? I thought by default integer division yields a float result.

**Comment by rpbouman** [comment]
Is this still the case? Just tested on v1.0.0 on plain SQL:

`SELECT CAST(1.0 AS HUGEINT) / CAST(2.0 AS HUGEINT);`
and the result is as desired:
```
┌───────────────────────────────────────────────┐
│ (CAST(1.0 AS HUGEINT) / CAST(2.0 AS HUGEINT)) │
│                    double                     │
├───────────────────────────────────────────────┤
│                                           0.5 │
└───────────────────────────────────────────────┘
```
(Consider closing the discussion if the problem is gone, alternatively file an issue for the Python interface)
